src/app/plot.py

```diff
@@ -0,0 +1,221 @@
+
+import numpy as np
+import pandas as pd
+import altair as alt
+import geopandas as gpd
+
+
+data = pd.read_csv("./../../data/processed/cleaned_salaries.csv")
+
+def plot_22(xmax, con):
+    
+    source = data[(data["Age"]>0) & (data["Salary_USD"]<=xmax[1])]
+    if con == "World" or con is None:
+        pass
+    else:
+        source = source[source["Country"] == con]
+
+    
+    alt.themes.enable("default")
+
+    chart = alt.Chart(source).mark_rect().encode(
+        x = alt.X("Age:Q", bin=alt.Bin(maxbins=60), title=None),
+        y = alt.Y("Salary_USD:Q", bin=alt.Bin(maxbins=40), 
+                  title="Salary in USD", axis=alt.Axis(format='~s')),
+        tooltip='count()',
+        color=alt.Color('count()',scale=alt.Scale(scheme='greenblue'), legend=alt.Legend(title='Tot. Records')),
+    ).properties(
+        title = "Heatmap of selected country",
+        width=335,
+        height=270,
+    )
+    
+    bar = alt.Chart(source).mark_bar().encode(
+        x='Age:Q',
+        y='count()',
+    ).properties(
+        width=335,
+        height=170,
+    )
+    
+    fchart = alt.vconcat(chart, bar, spacing=0)
+    
+    return fchart.to_html()
+
+
+def plot_12(xcon):
+    
+    genders = ['Male', 'Female', 'A different identity']
+    if xcon == "World" or xcon is None:
+        source = data[data["GenderSelect"].isin(genders)]
+    else:
+        source = data[(data["Country"] == xcon) & (data["GenderSelect"].isin(genders))]
```

Comment:
We actually have 4 categorical levels here: (1) Male, (2) Female, (3) Non-binary, genderqueer, or gender non-conforming, and (4) A different identity.

I think it would be better to group (3) and (4) into "A different identity" so we don't throw away any data.

Reply:
solved:Use replace